mro/table.py

```diff
@@ -98,14 +102,15 @@ def _convert_numpy_types_to_python(values):
         return values
 
     @classmethod
```

Comment:
I've added retries for all select operations. This protects us against the situation where `_execute_sql` runs ok but a disconnect occurs while the results are being fetched from the cursor. In this case, the entire select operation is retried, which will in turn re-run the query (which will trigger a reconnect).

We can only do this safely for `select` because it's idempotent. `insert`, `delete` and `update` remain as they are.